app/root.tsx

```diff
@@ -43,16 +56,39 @@ export const meta: MetaFunction = () => ({
 });
 
 type LoaderData = {
-    user: Awaited<ReturnType<typeof getUser>>;
+    user: User | null;
+    userSettings: UserSettings | null;
 };
 
 export const loader: LoaderFunction = async ({ request }) => {
+    const user = await getUser(request);
+    let userSettings: UserSettings | null = null;
```

Comment:
suggestion: Instead of using `null` like this special value we have to check for in our component, let's just default the settings here and they will never be `null`!